---
permalink: /cbio-who/
title: "Lab members"
excerpt: "The team"
modified: 2019-08-16
author_profile: false
sidebar:
  nav: "cbio"
---

{% include toc %}

## CBIO core

- [**Laurent Gatto**](http://lgatto.github.io/about/) (ORCID:
  [0000-0002-1520-2268](https://orcid.org/0000-0002-1520-2268),
  Github: [lgatto](https://github.com/lgatto/), Twitter
  [@lgatt0](https://twitter.com/lgatt0)), head of the unit. He runs
  the CBIO group and teaches bioinformatics in the Facutly of Pharmacy
  and Biomedical Sciences (FASB).

- **Axelle Loriot** (Github: [aloriot](https://github.com/aloriot)) is
  a senior researcher at the de Duve Institute. Since 2001, she has
  been part of the epigenetics group led by Charles De Smet. She
  joined the CBIO group in 2018, focusing on transcriptomics and
  epigenetics. She developed algorithms to identify aberrant
  transcripts induced by DNA hypomethylation in tumor cells, and DNA
  hypermethylation events linked to overlapping expression.

- **Chong Tang** (ORCID:
  [0000-0002-0482-8361](https://orcid.org/0000-0002-0482-8361),
  Github: [plantton](https://github.com/plantton)) is a PhD student
  working on mass spectrometry-based proteomics software
  development. He has a bachelor degree in biotechnology from the
  Central South University of Forestry and Technology in China and a
  masters in bioinformatics from KULeuven. He is funded by a China
  Scholarship Council - FNRS scholarship.

- **Philippe Hauchamps** (Github: [phauchamps](https://github.com/phauchamps))
  is a Master student in Biostatistics. He joined the CBIO lab in October 2019
  as an intern. He has a Master degree in Computer Science Engineering, has been
  previously working in the banking sector and is currently transitioning to
  Bioinformatics/Biostatistics.


## Remote members

- [**Lisa Breckels**](https://lmsimp.github.io/), Post-Doctoral
  Research Associa in the Cambridge Centre for Proteomics, at the
  University of Cambridge (UK). Her work focuses on the application
  and development of machine learning and pattern recognition methods
  to enable robust analysis of high-throughput quantitative proteomics
  data.  All her methods and algorithms form part of the
  [`pRoloc`](http://bioconductor.org/packages/pRoloc) framework for
  protein localisation prediction.

- [**Oliver Crook**](https://olivercrook.co.uk/) is a Wellcome Trust
  PhD student at the University of Cambridge, jointly supervised with
  [Paul
  Kirk](http://www.mrc-bsu.cam.ac.uk/people/in-alphabetical-order/h-to-m/paul-kirk/)
  from MRC Biostatistics Unit and [Kathryn
  Lilley](http://www.bioc.cam.ac.uk/people/uto/lilley). Olly works on
  new Bayesian methods for spatial proteomics ([Crook *et al.*
  2018](https://doi.org/10.1371/journal.pcbi.1006516)), formalising
  the theoretical foundations and implementing the algorithms that
  will be included in [`pRoloc`](https://lgatto.github.io/pRoloc/).

- **Aayush Grover** (ORCID:
  [0000-0002-3716-2850](https://orcid.org/0000-0002-3716-2850),
  Github: [AayushGrover](https://github.com/AayushGrover)) is a
  research intern at the CBIO lab. He has a bachelor's degree in
  Computer Science Engineering and is currently pursuing his Master's
  degree in Data Science from the International Institute of
  Information Technology Bangalore (India). He is currently working on
  predicting protein subcellular location using deep learning
  techniques.

## [R for Mass Spectrometry](https://www.rformassspectrometry.org/) core team

- [**Sebastian Gibb**](http://sebastiangibb.de/), Department of
  Anesthesiology and Intensive Care, University Medicine Greifswald,
  Germany. Sebastian visited Laurent's group in Cambridge 2014 and
  since then contributes to various projects, in particular
  [`MSnbase`](http://lgatto.github.io/MSnbase),
  [`Pbase`](http://bioconductor.org/packages/Pbase) and
  [`synapter`](http://bioconductor.org/packages/synapter).

- [**Johannes
  Rainer**](http://www.eurac.edu/en/research/health/biomed/staff/Pages/staffdetails.aspx?persId=34084),
  Center for Biomedicine, EURAC, Bolzano, Italy is a contributor to
  the [`MSnbase`](http://lgatto.github.io/MSnbase) and
  [`Pbase`](http://bioconductor.org/packages/Pbase) packages.

## Key collaborators

- [**Paul
  Kirk**](https://www.mrc-bsu.cam.ac.uk/people/in-alphabetical-order/h-to-m/paul-kirk/),
  MRC Biostatistics Unit, University of Cambridge, UK. Paul is part of
  the team developing principled Bayesian model for spatial
  proteomics, such as the T-Augmented Gaussian Mixture (TAGM) model
  ([Crook *et al.*
  2018](https://doi.org/10.1371/journal.pcbi.1006516)).

<!-- - [**Kathyn S. Lilley**](http://www.bioc.cam.ac.uk/people/uto/lilley), -->
<!--   Cambridge Centre for Proteomics, Department of Biochemistry, -->
<!--   University of Cambridge, UK. Kathryn's lab developed the LOPIT -->
<!--   ([Dunkley *et al.* -->
<!--   2006](https://www.ncbi.nlm.nih.gov/pubmed/16618929)) and hyperLOPIT -->
<!--   ([Christoforou *et al.* -->
<!--   2016](https://www.ncbi.nlm.nih.gov/pubmed/26754106)) -->
<!--   technologies. Her lab is at the origin of some of the very best -->
<!--   spatial proteomics datasets in the world, which we help to analyse -->
<!--   and interpret by applying our dedicated software (see for example -->
<!--   [Breckels *et al.* 2016 in -->
<!--   F1000Research](https://f1000research.com/articles/5-2926/)). -->

<!-- - [**Sean Holden**](http://www.cl.cam.ac.uk/~sbh11/), Computer -->
<!--   Laboratory. We collaborate with Sean on the development and -->
<!--   application of machine learning algorithms to solve problems in -->
<!--   spatial proteomics. A recent example is transfer learning -->
<!--   ([Breckels *et al.* 2016](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004920)). -->

- [**Thomas
  Burger**](https://sites.google.com/site/thomasburgerswebpage/),
  BIG-BGE (Universit√© Grenoble Alpes, CNRS, CEA, INSERM), France. We
  collaborate with Thomas and his group on various proteomics data
  analysis (for example missing data in [Lazar *et al*
  2016](https://www.ncbi.nlm.nih.gov/pubmed/26906401)) and software
  (see [Wieczorek *et al.*
  2016](https://www.ncbi.nlm.nih.gov/pubmed/27605098)). Thomas
  contributed the *perTurbo* algorithms to
  [*pRoloc*](https://bioconductor.org/packages/pRoloc).

- [**Lieven Clement**](https://statomics.github.io/), statOmics group,
  UGent, Belgium. We collaborate on statistical analyses of
  quantitative proteomics data.

- [**Christophe Vanderaa**](https://github.com/cvanderaa), statOmics group,
  UGent, Belgium. We pursue our collaboration on the development of
  computational solutions for the analysis of single-cell proteomics
  data.

## Alumni

- **Christophe Vanderaa** (ORCID:
  [0000-0001-7443-5427](https://orcid.org/0000-0001-7443-5427),
  Github: [cvanderaa](https://github.com/cvanderaa)) is a former PhD
  student working on single-cell assay bioinformatics. He did a
  masters in biomedical sciences at the UCLouvain and followed up with
  a masters in bioinformatics at the KULeuven before conducting his
  PhD training in the lab funded by an FNRS fellowship.

- [**Theo Killian**](https://github.com/tfkillian) was a
  bioinformatician and research programmer in the CBIO group at the de
  Duve Institute. Theo developes and maintains the
  [depmap](http://bioconductor.org/packages/release/data/experiment/html/depmap.html)
  package.

- [**Robin Kohze**](https://github.com/Kohze) comes from the Radboud
  University Nijmegen and visits the group in the frame of his
  master's internship. He developed a new infrastructure to facilitate
  the dissemination of spatial proteomics data. He currently pursues a
  PhD in the Department of Genetics at the University of Cambridge,
  UK.

- **Thomas Naake**, visited the group in Spring 2014 as an ERASMUS
  student affiliated to the University of Freiburg. He developed the
  first version of
  [`pRolocGUI`](https://bioconductor.org/packages/release/bioc/html/pRolocGUI.html),
  an interactive visualisation tools for organelle proteomics
  data. Thomas is now a PhD student at Max Planck Institute of
  Molecular Plant Physiology.

- **Victoria Carr**, Part III student in Systems Biology, applying
  graph-based methods to study protein sub-cellular
  localisation. Victoria is now pursuing a PhD at King's College
  London.
